src/org/rust/lang/core/psi/impl/mixin/RustPathPartImplMixin.kt

```diff
@@ -24,6 +25,30 @@ abstract class RustPathPartImplMixin(node: ASTNode) : RustNamedElementImpl(node)
             if (it.firstChild != null) it else null
         }
 
+    private val isViewPath: Boolean
+        get() = when (parent) {
+            is RustViewPath          -> true
+            is RustPathPartImplMixin -> (parent as RustPathPartImplMixin).isViewPath
```

Comment:
No, because parent is a getter. I've hit this problem couple of times already :(

Other possible workarounds are `val parent = parent` or `parent.let {parent ->}`.